### PR TITLE
Redesign Assessment Module

### LINKED ISSUE #5912: Assesment Reports
- [ ] Individual Student assessment.
- [ ] Class assessment (course wise).
- [ ] Course assessment (group wise).
- [ ] Assessment group report.

### PR DESCRIPTION
**Pull request by neilLasrado**
Fixes #5912

### CODE REVIEW

File: erpnext/schools/api.py [hunk 1]
```diff
@@ -163,3 +163,41 @@ def get_course_schedule_events(start, end, filters=None):
...
+	"""Returns Evaluation Criteria  and Maximum Score from Assessment Master.
+
+	:param Assessment: Assessment
+	"""
+	return frappe.get_list("Assessment Evaluation Criteria", \
+		fields=["evaluation_criteria", "maximum_score"], filters={"parent": assessment}, order_by= "idx")
+
+
+@frappe.whitelist()
+def get_grade(grading_scale, percentage):
```
Comment (neilLasrado) [inline erpnext/schools/api.py]: Check this function

Comment (neilLasrado) [inline erpnext/schools/api.py]: Check this Function

File: erpnext/patches/v7_2/update_assessment_modules.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import frappe
+from frappe.model.utils.rename_field import rename_field
+
+def execute():
```
Comment (neilLasrado) [inline erpnext/patches/v7_2/update_assessment_modules.py]: Check this patch.

File: erpnext/schools/doctype/course/course.py [hunk 3]
```diff
@@ -8,7 +8,16 @@
...
+	def validate_evaluation_criterias(self):
+		if self.evaluation_criterias:
+			total_weightage = 0
+			for criteria in self.evaluation_criterias:
+				total_weightage += criteria.weightage
+			if total_weightage != 100:
+				frappe.throw(_("Total Weightage of all Evaluation Criterias must be 100%"))
 
 def get_sg_list(doctype, txt, filters, limit_start, limit_page_length=20):
 	user = frappe.session.user
```
Comment (rmehta) [inline erpnext/schools/doctype/course/course.py]: bad function name, don't use abbreviations

File: erpnext/schools/doctype/assessment_result/assessment_result.py [hunk 4]
```diff
@@ -4,7 +4,22 @@
...
+	def validate_grade(self):
+		self.total_score = 0.0
+		for d in self.details:
+			if d.score > d.maximum_score:
+				frappe.throw(_("Score cannot be greater than Maximum Score"))
+			else:
+				d.grade = get_grade(self.grading_scale, (flt(d.score)/d.maximum_score)*100)
+				self.total_score += d.score
+		self.grade = get_grade(self.grading_scale, (self.total_score/self.maximum_score)*100)
+
```
Comment (rmehta) [inline erpnext/schools/doctype/assessment_result/assessment_result.py]: test case?

**Comment by neilLasrado** [comment]
Screenshots- 

### Evaluation Criteria:

screen shot 2016-12-30 at 4 44 42 pm

### Grading Scale:

screen shot 2016-12-30 at 4 44 21 pm

### Course Master:

screen shot 2016-12-30 at 4 45 10 pm

### Assessment:

screen shot 2016-12-30 at 4 43 10 pm

### Assessment Result:

screen shot 2016-12-30 at 4 43 40 pm

File: erpnext/schools/doctype/course/course.py (hunk 3 above)
Comment (neilLasrado) [inline erpnext/schools/doctype/course/course.py]: The function has been removed. Was LMS related code.

**Review by nabinhait — APPROVED** [review]

**Pull request merged**